### ISSUE TITLE
Fixes to jump

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#645]
- [x] Have jump check that `to > self.ind`
- [x] Have `jump_dt()` take an arg

**Comment by cliffckerr** [comment]
There's almost no difference for the baseline sim, where networks are regenerated on each timestep, so the probability of getting two draws from the same agents on two consecutive timesteps are limited. I would imagine there's a bigger impact with sexual networks, where the same edges are tested repeatedly. But even then the differences seem to be small, roughly comparable to using a different random seed:
```py
import sciris as sc
import matplotlib.pyplot as plt

# Import the two versions
parent = sc.path('/home/cliffk/idm/')
folder1 = parent / 'starsim/starsim' # Verison with temporary fix
folder2 = parent / 'starsim2/starsim' # Main; NB, you may need to delete __pycache__ for this to work
ss1 = sc.importbypath(folder1)
ss2 = sc.importbypath(folder2)

# Define the parameters
pars = sc.objdict(
    n_agents   = 100e3, # Number of agents
    start      = 2000, # Starting year
    n_years    = 20,   # Number of years to simulate
    dt         = 0.2,  # Timestep
    rand_seed  = 2,    # Set a non-default seed
    diseases = ['hiv', 'syphilis'],
    networks = ['mf', 'maternal'],
    demographics = True,
)

# Create and run the sims
sim1 = ss1.Sim(pars, label='With fix')
sim2 = ss2.Sim(pars, label='With bug')
for sim in [sim1, sim2]:
    sim.run()
    sim.flat = sim.results.flatten()
    yearvec = sim.flat.pop('yearvec')
    for k,v in list(sim.flat.items()):
        if v.std() == 0 or k in ['new_deaths', 'cum_deaths']:
            sim.flat.pop(k) # Don't need them if no values

# Plot
n = len(sim.flat)
nrows,ncols = sc.getrowscols(n)
for i in range(n):
    plt.subplot(nrows, ncols, i+1)
    for sim in [sim1, sim2]:
        res = sim.flat[i]
        plt.plot(yearvec, res, label=sim.label, alpha=0.7)
    plt.title(sim.flat.keys()[i])
    plt.legend()
    
sc.figlayout()
plt.show()
```
![image](https://github.com/user-attachments/assets/233d9b1b-545e-4b37-9146-b42af19f1188)

**Comment by cliffckerr** [comment]
Closed by #647
